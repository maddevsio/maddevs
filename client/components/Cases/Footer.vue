<template>
  <section class="case_footer container_full">
    <slot name="icon"></slot>
    <h2 :class="`case_title-${className}`" class="case_title case_title_h2">
      <slot></slot>
    </h2>
    <a :href="link" class="case_next-project-button">Read next project</a>
  </section>
</template>

<script>
export default {
  name: 'Header',
  props: {
    link: {
      type: String,
      default: '',
    },

    className: {
      type: String,
      default: '',
    },
  },
}
</script>

<style scoped lang="scss">
@import '../../assets/styles/cases/mixins';

.case {
  &_footer {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 85px;
    padding-bottom: 115px;
    position: relative;
    background-color: $bgcolor--white-transparent;

    &::before {
      content: '';
      width: calc(100% - 198px);
      height: 2px;
      position: absolute;
      top: 0;
      background-color: $border-color--grey-cases;
    }
  }

  &_title {
    text-align: center;
  }

  &_title-sjmc-case,
  &_title-godee-case {
    margin-bottom: 51px;
  }

  &_title-namba-food-case {
    margin-bottom: 25px;
  }

  &_next-project-button {
    padding: 11px 28px;
    background-color: $bgcolor--black;
    border: 0;
    cursor: pointer;
    text-decoration: none;
    @include default_text($text-color--white-primary, 17px, 166%, -0.035em, normal);
  }

  @media screen and (max-width: 768px) {
    &_footer {
      padding-top: 70px;
      padding-bottom: 100px;

      &::before {
        width: calc(100% - 99px);
      }
    }

    &_title-sjmc-case,
    &_title-godee-case {
      margin-bottom: 41px;
    }

    &_next-project-button {
      @include default_text($text-color--white-primary, 16px, 150%, -0.02em, normal);
    }
  }
}
</style>
