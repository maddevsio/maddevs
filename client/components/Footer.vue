<template>
  <footer :class="`footer ${currentPageName}`">
    <div class="container">
      <div class="footer__content row">
        <div class="footer__left-content col-xl-8 col-lg-7 col-md-6">
          <h2 class="footer__main-title">Get In Touch</h2>
          <footerContacts />
          <footerNavbar class="footer__navbar-desktop" />
        </div>
        <div class="footer__form-wrap col-xl-4 col-lg-5 col-md-5">
          <footerForm />
        </div>
      </div>
      <div class="footer__mobile-content footer__social-network-list-mobile">
        <footerSocialNetWorkList />
      </div>
      <div class="footer__mobile-content">
        <footerNavbarMobile />
      </div>
    </div>
  </footer>
</template>

<script>
import footerForm from '@/components/Footer/footer-form'
import footerContacts from '@/components/Footer/footer-contacts'
import footerNavbar from '@/components/Footer/footer-navbar'
import footerNavbarMobile from '@/components/Footer/footer-navbar-mobile'
import footerSocialNetWorkList from '@/components/Footer/footer-social-network-list'

export default {
  name: 'Footer',
  components: {
    footerForm,
    footerContacts,
    footerNavbar,
    footerNavbarMobile,
    footerSocialNetWorkList,
  },

  data() {
    return {
      currentYear: new Date().getFullYear(),
      currentPageName: '',
    }
  },

  watch: {
    $route() {
      this.updateClassName()
    },
  },

  mounted() {
    this.updateClassName()
  },

  methods: {
    updateClassName() {
      if ($nuxt.$route.name) {
        this.currentPageName = $nuxt.$route.name
      }
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../assets/styles/vars';

.footer {
  padding-top: 100px;
  padding-bottom: 92px;

  &__content {
    justify-content: space-between;

    @media screen and (max-width: 991px) {
      margin: 0;
    }
  }

  &__main-title {
    margin-bottom: 43px;
    font-size: 62px;
    font-family: 'Poppins-Bold', sans-serif;
    font-weight: 700;
    line-height: 120%;
    letter-spacing: -1px;
    color: $text-color--red;
  }

  &__left-content {
    padding-right: 128px;
    box-sizing: border-box;
  }

  &__form-wrap {
    margin-top: 20px;
  }

  &__mobile-content {
    display: none;
  }

  &__social-network-list-mobile {
    margin: 40px 0;
  }
}

.gdpr,
.privacy,
.nda {
  margin-top: 141px;
}

.gdpr,
.privacy,
.nda,
.faq {
  padding-top: 109px;
  border-top: 1px solid $border-color--grey-dark;
}

.faq {
  margin-top: 114px;
}

@media only screen and (max-width: 1200px) {
  .footer {
    &__main-title {
      margin-bottom: 20px;
    }

    &__form-wrap {
      margin-top: 33px;
    }

    &__left-content {
      padding-right: 40px;
    }
  }
}

@media screen and (max-width: 990px) {
  .footer__main-title {
    font-size: 45px;
  }
}

@media only screen and (max-width: 991px) {
  .footer {
    padding-top: 48px;
    padding-bottom: 46px;

    &__form-wrap,
    &__left-content {
      padding-left: 0;
      padding-right: 0;
    }

    &__form-wrap {
      margin-top: 30px;
    }
  }
}

@media only screen and (max-width: 834px) {
  .gdpr,
  .privacy,
  .nda {
    margin-top: 90px;
  }

  .faq {
    margin-top: 65px;
  }
}

@media screen and (max-width: 767px) {
  .footer {
    &__main-title {
      font-size: 45px;
    }

    &__navbar-desktop {
      display: none;
    }

    &__mobile-content {
      display: block;
    }

    &__social-network-list-mobile {
      border-top: 1px solid rgba(164, 168, 180, 0.2);
      padding-top: 35px;
    }
  }
}

@media only screen and (max-width: 578px) {
  .gdpr,
  .privacy,
  .nda {
    margin-top: 68px;
  }

  .faq {
    margin-top: 44px;
  }
}

@media only screen and (max-width: 576px) {
  .footer {
    padding-top: 50px;

    &__main-title {
      font-size: 46px;
    }

    &__social-network-list-mobile {
      margin-left: 0;
      margin-right: 0;
    }
  }
}
</style>
