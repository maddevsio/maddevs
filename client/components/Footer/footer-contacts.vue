<template>
  <div class="footer-contacts">
    <div class="footer-contacts__head-content">
      <div class="footer-contacts__contact-item">
        <p class="footer-contacts__contact-title">Text us:</p>
        <a :href="`mailto:${mailLink}`" class="footer-contacts__contact-link footer-contacts__contact-mail">{{
          mailLink
        }}</a>
      </div>
      <div class="footer-contacts__contact-item">
        <div class="footer-contacts__contact-title-wrapper">
          <div class="footer-contacts__contact-title-flag">
            <img
              :data-src="require(`@/assets/img/Home/flags/uk.svg`)"
              width="19"
              height="14"
              class="img_lazy"
              alt="United Kingdom"
            />
          </div>
          <p class="footer-contacts__contact-title">Call us:</p>
        </div>
        <a href="tel:+442039848555" class="footer-contacts__contact-link footer-contacts__contact-phone-number"
          >+44 20 3984 8555</a
        >
      </div>
    </div>
    <div class="footer-contacts__lists-wrapper">
      <ul class="footer-contacts__messengers-list">
        <li>
          <a href="https://m.me/maddevsllc" target="_blank" class="footer-contacts__messenger-item-wrapper">
            <img
              :data-src="require(`@/assets/img/Footer/svg/messenger.svg`)"
              width="42"
              height="42"
              class="img_lazy"
              alt="Messenger"
            />
            <p class="footer-contacts__messenger-name">Messenger</p>
          </a>
        </li>
        <li>
          <a
            href="https://wa.me/message/KPJAW6J7BF7EK1"
            target="_blank"
            class="footer-contacts__messenger-item-wrapper"
          >
            <img
              :data-src="require(`@/assets/img/Footer/svg/whatsapp.svg`)"
              width="42"
              height="42"
              class="img_lazy"
              alt="WhatsApp"
            />
            <p class="footer-contacts__messenger-name">WhatsApp</p>
          </a>
        </li>
        <li>
          <a href="https://t.me/MadDevs_io" target="_blank" class="footer-contacts__messenger-item-wrapper">
            <img
              :data-src="require(`@/assets/img/Footer/svg/telegram.svg`)"
              width="42"
              height="42"
              class="img_lazy"
              alt="Telegram"
            />
            <p class="footer-contacts__messenger-name">Telegram</p>
          </a>
        </li>
        <li>
          <a
            href="https://msng.link/o/?https%3A%2F%2Fu.wechat.com%2FICWluRgJH8tu0IisMQ1eEFo=wc"
            target="_blank"
            class="footer-contacts__messenger-item-wrapper"
          >
            <img
              :data-src="require(`@/assets/img/Footer/svg/wechat.svg`)"
              width="42"
              height="42"
              class="img_lazy"
              alt="WeChat"
            />
            <p class="footer-contacts__messenger-name">WeChat</p>
          </a>
        </li>
      </ul>
      <div class="footer-contacts__social-network-list-desktop">
        <footerSocialNetworkList />
      </div>
    </div>
  </div>
</template>

<script>
import footerSocialNetworkList from '@/components/Footer/footer-social-network-list'

export default {
  name: 'FooterContacts',
  components: {
    footerSocialNetworkList,
  },

  data: () => ({
    mailLink: process.env.emailContact,
  }),
}
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';

@mixin contacts-head-content {
  &__head-content {
    flex-direction: column;
    margin-bottom: 31px;
  }
}

@mixin messenger-list-grid {
  &__messenger-item-wrapper {
    justify-content: flex-start;
  }

  &__messenger-item-wrapper {
    &:nth-child(3) {
      margin-left: -8px;
    }
  }
}

.footer-contacts {
  &__head-content {
    margin-bottom: 68px;
  }

  &__head-content,
  &__messengers-list,
  &__messenger-item-wrapper {
    display: flex;
    justify-content: space-between;
  }

  &__contact-title-wrapper {
    display: flex;
    align-items: center;
  }

  &__contact-title,
  &__messenger-name {
    font-size: 16px;
    line-height: 24px;
    letter-spacing: -0.02em;
    font-family: 'Poppins-Regular', sans-serif;
  }

  &__contact-title-flag {
    display: flex;
    align-items: center;
    margin-right: 8px;

    svg {
      width: 18px;
      height: 14px;
    }
  }

  &__contact-title,
  &__contact-link {
    color: $text-color--white;
  }

  &__contact-link {
    display: inline-block;
    margin-top: 12px;
    font-size: 34px;
    line-height: 100%;
    font-weight: 600;
    font-family: 'Poppins-Medium', sans-serif;
    letter-spacing: -1px;
    border-bottom: 1px solid $text-color--red-opacity;
    text-decoration: none;
  }

  &__messenger-item-wrapper {
    align-items: center;
    text-decoration: none;

    &:first-child {
      margin-left: -8px;
    }

    img {
      display: block;
      width: 42px;
      height: 42px;
      margin-right: 3px;
    }
  }

  &__messenger-name {
    color: $text-color--grey;
  }

  &__social-network-list-desktop {
    margin-top: 53px;
    padding-top: 53px;
    border-top: 1px solid $border-color--grey-dark-transparent;
  }
}

@media only screen and (max-width: 1320px) {
  .footer-contacts {
    &__contact-link {
      font-size: 28px;
    }
  }
}

@media only screen and (min-width: 768px) and (max-width: 1200px) {
  .footer-contacts {
    &__contact-item {
      margin-bottom: 42px;

      &:last-child {
        margin-top: 0;
        margin-bottom: 0;
      }
    }

    &__social-network-list-desktop {
      margin-top: 35px;
      padding-top: 40px;
    }

    @include contacts-head-content;
    @include messenger-list-grid;
  }
}

@media screen and (max-width: 1100px) {
  .footer-contacts__messengers-list {
    width: max-content;
    display: grid;
    grid-template-columns: repeat(2, max-content);
    grid-column-gap: 70px;
  }
}

@media only screen and (max-width: 991px) {
  .footer-contacts {
    &__head-content {
      flex-direction: row;
      margin-bottom: 48px;
      flex-wrap: wrap;
    }

    &__lists-wrapper {
      max-width: 100%;
    }
  }
}

@media screen and (max-width: 800px) {
  .footer-contacts__messengers-list {
    grid-column-gap: 60px;
  }
}

@media screen and (max-width: 767px) {
  .footer-contacts {
    &__social-network-list-desktop {
      display: none;
    }

    &__contact-item {
      &:last-child {
        margin-top: 0;
        margin-bottom: 0;
      }
    }

    &__messenger-telegram {
      margin-left: 2px;
    }
  }
}

@media only screen and (max-width: 640px) {
  .footer-contacts {
    @include contacts-head-content;
    @include messenger-list-grid;

    &__contact-item {
      margin-bottom: 33px;
    }

    &__contact-title {
      font-size: 13px;
      line-height: 22px;
    }
  }
}

@media screen and (max-width: 600px) {
  .footer-contacts {
    &__messengers-list {
      width: max-content;
      display: grid;
      grid-template-columns: repeat(2, max-content);
      grid-column-gap: 70px;

      svg {
        width: 42px;
        height: 42px;
      }
    }
  }
}

@media screen and (max-width: 360px) {
  .footer-contacts__messengers-list {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(2, max-content);
    grid-column-gap: 33px;
  }
}
</style>
