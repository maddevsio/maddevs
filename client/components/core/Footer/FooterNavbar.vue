<template>
  <div class="footer-navbar">
    <p class="footer-navbar__company-name">
      Â© Mad Devs - {{ currentYear }}
    </p>
    <div class="footer-navbar__nav-list">
      <NuxtLink
        v-for="item in navigation"
        :key="item.link"
        :to="item.link"
        target="_blank"
        class="footer-navbar__nav-item"
      >
        {{ item.title }}
      </NuxtLink>
    </div>
  </div>
</template>

<script>
import { footerNavigation as navigation } from '@/data/navigation'

export default {
  name: 'FooterNavbar',
  data() {
    return {
      navigation,
      currentYear: new Date().getFullYear(),
    }
  },
}
</script>

<style lang="scss" scoped>
@import '../../../assets/styles/vars';

.footer-navbar {
  width: max-content;
  margin-top: 64px;

  &__nav-list {
    display: flex;
  }

  &__nav-item,
  &__company-name {
    font-size: 16px;
    line-height: 24px;
    color: $text-color--grey;
    letter-spacing: -0.02em;
  }

  &__company-name {
    margin-bottom: 6px;
  }

  &__nav-item {
    margin-right: 26px;
    text-decoration: underline;
    text-decoration-color: $footer--text-decoration-color;

    &:last-child {
      margin-right: 0;
    }
  }
}

@media only screen and (max-width: 1320px) {
  .footer-navbar {
    width: initial;
    margin-top: 40px;

    &__nav-list {
      flex-wrap: wrap;
    }

    &__nav-item,
    &__company-name {
      margin-bottom: 3px;
    }

    &__nav-item {
      &:nth-child(1) {
        width: 100%;
      }

      &:nth-child(2),
      &:nth-child(3) {
        margin-right: 15px;
      }

      &:nth-child(4) {
        margin-left: 0;
        margin-bottom: 0;
      }
    }
  }
}

@media only screen and (max-width: 991px) {
  .footer-navbar {
    &__nav-list {
      flex-direction: column;
    }
  }
}
</style>
