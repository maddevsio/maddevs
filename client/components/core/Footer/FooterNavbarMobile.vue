<template>
  <div class="footer-navbar-mobile">
    <p class="footer-navbar-mobile__company-name">
      Â© Mad Devs - {{ currentYear }}
    </p>
    <div class="footer-navbar-mobile__nav-list">
      <NuxtLink
        v-for="item in navigation"
        :key="item.link"
        :to="item.link"
        target="_blank"
        class="footer-navbar-mobile__nav-link"
      >
        {{ item.title }}
      </NuxtLink>
    </div>
  </div>
</template>

<script>
import { footerNavigation as navigation } from '@/data/navigation'

export default {
  name: 'FooterNavbarMobile',
  data() {
    return {
      navigation,
      currentYear: new Date().getFullYear(),
    }
  },
}
</script>

<style lang="scss" scoped>
.footer-navbar-mobile {
  margin-top: 40px;
  text-align: center;

  &__nav-list {
    display: flex;
    flex-direction: column;
  }

  &__nav-link,
  &__company-name {
    font-size: 16px;
    color: $text-color--grey;
    letter-spacing: -0.02em;
    text-decoration: none;
  }

  @media only screen and (max-width: 470px) {
    text-align: left;
  }
}
</style>
