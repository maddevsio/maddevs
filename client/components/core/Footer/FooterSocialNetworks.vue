<template>
  <ul class="footerSocialNetworkList">
    <li
      v-for="network in socialNetworks"
      :key="network.key"
      class="footerSocialNetworkList__social-network-link-wrapper"
    >
      <a
        :href="network.url"
        class="footerSocialNetworkList__social-network-link"
        target="_blank"
      >
        <img
          :data-src="require(`@/assets/img/Footer/svg/${network.key}.svg`)"
          :alt="network.title"
          width="42"
          height="42"
          class="img_lazy"
        >
      </a>
    </li>
  </ul>
</template>

<script>
import { socialNetworks } from '@/data/socialNetworks'

export default {
  name: 'FooterSocialNetworkList',
  data() {
    return {
      socialNetworks,
    }
  },
}
</script>

<style lang="scss" scoped>
@mixin social-network-list-grid {
  .footerSocialNetworkList {
    display: grid;
    grid-template-columns: repeat(4, max-content);

    &__social-network-link-wrapper {
      margin: 0 -8px;
    }
  }
}

.footerSocialNetworkList {
  display: flex;
  justify-content: space-between;

  &__social-network-link {
    width: 42px;
    height: 42px;
    display: block;

    img {
      width: 42px;
      height: 42px;
      display: block;
      overflow: hidden;
    }
  }
}

@media screen and (min-width: 768px) and (max-width: 1200px) {
  @include social-network-list-grid;
}

@media screen and (max-width: 640px) {
  @include social-network-list-grid;
}
</style>
