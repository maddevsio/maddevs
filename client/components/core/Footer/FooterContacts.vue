<template>
  <div class="footer-contacts">
    <div class="footer-contacts__head-content">
      <div class="footer-contacts__contact-item">
        <p class="footer-contacts__contact-title">
          Text us:
        </p>
        <a
          :href="`mailto:${mailLink}`"
          class="footer-contacts__contact-link footer-contacts__contact-mail"
        >{{
          mailLink
        }}</a>
      </div>
      <div class="footer-contacts__contact-item">
        <div class="footer-contacts__contact-title-wrapper">
          <div class="footer-contacts__contact-title-flag">
            <img
              :data-src="require(`@/assets/img/Home/flags/uk.svg`)"
              width="19"
              height="14"
              class="img_lazy"
              alt="United Kingdom"
            >
          </div>
          <p class="footer-contacts__contact-title">
            Call us:
          </p>
        </div>
        <a
          href="tel:+442039848555"
          class="footer-contacts__contact-link footer-contacts__contact-phone-number"
        >+44 20 3984 8555</a>
      </div>
    </div>
    <div class="footer-contacts__lists-wrapper">
      <ul class="footer-contacts__messengers-list">
        <li
          v-for="messenger in messengers"
          :key="messenger.key"
        >
          <a
            :href="messenger.url"
            target="_blank"
            class="footer-contacts__messenger-item-wrapper"
          >
            <img
              :data-src="require(`@/assets/img/Footer/svg/${messenger.key}.svg`)"
              width="42"
              height="42"
              class="img_lazy"
              :alt="messenger.label"
            >
            <p class="footer-contacts__messenger-name">{{ messenger.label }}</p>
          </a>
        </li>
      </ul>
      <div class="footer-contacts__social-network-list-desktop">
        <FooterSocialNetworks />
      </div>
    </div>
  </div>
</template>

<script>
import { messengers } from '@/data/messengers'
import FooterSocialNetworks from '@/components/core/Footer/FooterSocialNetworks'

export default {
  name: 'FooterContacts',
  components: {
    FooterSocialNetworks,
  },

  data() {
    return {
      messengers,
      mailLink: process.env.emailContact,
    }
  },
}
</script>

<style lang="scss" scoped>
@mixin contacts-head-content {
  &__head-content {
    flex-direction: column;
    margin-bottom: 31px;
  }
}

@mixin messenger-list-grid {
  &__messenger-item-wrapper {
    justify-content: flex-start;
  }

  &__messenger-item-wrapper {
    &:nth-child(3) {
      margin-left: -8px;
    }
  }
}

.footer-contacts {
  &__head-content {
    margin-bottom: 68px;
  }

  &__head-content,
  &__messengers-list,
  &__messenger-item-wrapper {
    display: flex;
    justify-content: space-between;
  }

  &__contact-title-wrapper {
    display: flex;
    align-items: center;
  }

  &__contact-title,
  &__messenger-name {
    @include font('Inter', 16px, 400);
    line-height: 24px;
    letter-spacing: -0.013em;
  }

  &__contact-title-flag {
    display: flex;
    align-items: center;
    margin-right: 8px;

    svg {
      width: 18px;
      height: 14px;
    }
  }

  &__contact-title,
  &__contact-link {
    color: $text-color--white-primary;
  }

  &__contact-link {
    @include font('Poppins', 34px, 600);
    display: inline-block;
    margin-top: 12px;
    line-height: 100%;
    letter-spacing: -1px;
    border-bottom: 1px solid $text-color--red-opacity;
    text-decoration: none;
  }

  &__messenger-item-wrapper {
    align-items: center;
    text-decoration: none;

    &:first-child {
      margin-left: -8px;
    }

    img {
      display: block;
      width: 42px;
      height: 42px;
      margin-right: 3px;
    }
  }

  &__messenger-name {
    color: $text-color--grey;
  }

  &__social-network-list-desktop {
    margin-top: 53px;
    padding-top: 53px;
    border-top: 1px solid $border-color--grey-dark-transparent;
  }
}

@media only screen and (max-width: 1320px) {
  .footer-contacts {
    &__contact-link {
      font-size: 28px;
    }
  }
}

@media only screen and (min-width: 768px) and (max-width: 1200px) {
  .footer-contacts {
    &__contact-item {
      margin-bottom: 42px;

      &:last-child {
        margin-top: 0;
        margin-bottom: 0;
      }
    }

    &__social-network-list-desktop {
      margin-top: 35px;
      padding-top: 40px;
    }

    @include contacts-head-content;
    @include messenger-list-grid;
  }
}

@media screen and (max-width: 1100px) {
  .footer-contacts__messengers-list {
    width: max-content;
    display: grid;
    grid-template-columns: repeat(2, max-content);
    grid-column-gap: 70px;
  }
}

@media only screen and (max-width: 991px) {
  .footer-contacts {
    &__head-content {
      flex-direction: row;
      margin-bottom: 48px;
      flex-wrap: wrap;
    }

    &__lists-wrapper {
      max-width: 100%;
    }
  }
}

@media screen and (max-width: 800px) {
  .footer-contacts__messengers-list {
    grid-column-gap: 60px;
  }
}

@media screen and (max-width: 767px) {
  .footer-contacts {
    &__social-network-list-desktop {
      display: none;
    }

    &__contact-item {
      &:last-child {
        margin-top: 0;
        margin-bottom: 0;
      }
    }

    &__messenger-telegram {
      margin-left: 2px;
    }
  }
}

@media only screen and (max-width: 640px) {
  .footer-contacts {
    @include contacts-head-content;
    @include messenger-list-grid;

    &__contact-item {
      margin-bottom: 33px;
    }

    &__contact-title {
      font-size: 13px;
      line-height: 22px;
    }
  }
}

@media screen and (max-width: 600px) {
  .footer-contacts {
    &__messengers-list {
      width: max-content;
      display: grid;
      grid-template-columns: repeat(2, max-content);
      grid-column-gap: 70px;

      svg {
        width: 42px;
        height: 42px;
      }
    }
  }
}

@media screen and (max-width: 360px) {
  .footer-contacts__messengers-list {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(2, max-content);
    grid-column-gap: 33px;
  }
}
</style>
