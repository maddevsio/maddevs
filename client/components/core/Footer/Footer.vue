<template>
  <footer
    data-testid="test-footer"
    :class="`footer ${pageName}`"
  >
    <div class="container">
      <div class="footer__content">
        <div class="footer__content-left">
          <h2 class="footer__main-title">
            Get In Touch
          </h2>
          <FooterContacts />
          <FooterNavbar class="footer__navbar-desktop" />
        </div>
        <div class="footer__content-right">
          <FooterForm />
        </div>
      </div>
      <div class="footer__mobile-content footer__social-network-list-mobile">
        <FooterSocialNetworks />
      </div>
      <div class="footer__mobile-content">
        <FooterNavbarMobile />
      </div>
    </div>
  </footer>
</template>

<script>
import FooterForm from '@/components/core/forms/FooterForm'
import FooterContacts from '@/components/core/Footer/FooterContacts'
import FooterNavbar from '@/components/core/Footer/FooterNavbar'
import FooterNavbarMobile from '@/components/core/Footer/FooterNavbarMobile'
import FooterSocialNetworks from '@/components/core/Footer/FooterSocialNetworks'

export default {
  name: 'Footer',
  components: {
    FooterForm,
    FooterContacts,
    FooterNavbar,
    FooterNavbarMobile,
    FooterSocialNetworks,
  },

  data() {
    return {
      pageName: '',
    }
  },

  watch: {
    $route() {
      this.updateClassName()
    },
  },

  mounted() {
    this.updateClassName()
  },

  methods: {
    updateClassName() {
      if ($nuxt.$route.name) {
        this.pageName = $nuxt.$route.name
      }
    },
  },
}
</script>

<style lang="scss" scoped>
.footer {
  padding-top: 100px;
  padding-bottom: 92px;
  @media only screen and (max-width: 991px) {
    padding-top: 48px;
    padding-bottom: 46px;
  }

  &__content {
    display: grid;
    grid-template-columns: 2fr 1fr;
    margin-top: 20px;

    @media screen and (max-width: 1200px) {
      grid-template-columns: 7fr 5fr;
    }

    @media screen and (max-width: 992px) {
      grid-template-columns: 6fr 5fr;
    }

    @media screen and (max-width: 768px) {
      grid-template-columns: 1fr;
    }

    &-left {
      padding-right: 128px;
      box-sizing: border-box;
      @media only screen and (max-width: 1200px) {
        padding-right: 40px;
      }
      @media only screen and (max-width: 991px) {
        padding-left: 0;
        padding-right: 0;
      }
    }
    &-right {
      margin-top: 20px;
      @media only screen and (max-width: 1200px) {
        margin-top: 33px;
      }
      @media only screen and (max-width: 991px) {
        padding-left: 0;
        padding-right: 0;
        margin-top: 10px;
        margin-left: 30px;
      }
      @media only screen and (max-width: 768px) {
        margin-left: 0;
        margin-top: 30px;
      }
    }
  }

  &__main-title {
    @include font('Poppins', 62px, 700);
    margin-bottom: 43px;
    line-height: 120%;
    letter-spacing: -1px;
    color: $text-color--red;
  }

  &__mobile-content {
    display: none;
  }

  &__social-network-list-mobile {
    margin: 40px 0;
  }
}

.gdpr,
.privacy,
.nda {
  margin-top: 141px;
}

.gdpr,
.privacy,
.nda,
.faq {
  padding-top: 109px;
  border-top: 1px solid $border-color--grey-dark;
}

.faq {
  margin-top: 114px;
}

@media only screen and (max-width: 1200px) {
  padding-right: 40px;
}

@media only screen and (max-width: 1200px) {
  .footer {
    &__main-title {
      margin-bottom: 20px;
    }
  }
}

@media screen and (max-width: 990px) {
  .footer__main-title {
    font-size: 45px;
  }
}

@media only screen and (max-width: 834px) {
  .gdpr,
  .privacy,
  .nda {
    margin-top: 90px;
  }

  .faq {
    margin-top: 65px;
  }
}

@media screen and (max-width: 767px) {
  .footer {
    &__main-title {
      font-size: 45px;
    }

    &__navbar-desktop {
      display: none;
    }

    &__mobile-content {
      display: block;
    }

    &__social-network-list-mobile {
      border-top: 1px solid rgba(164, 168, 180, 0.2);
      padding-top: 35px;
    }
  }
}

@media only screen and (max-width: 578px) {
  .gdpr,
  .privacy,
  .nda {
    margin-top: 68px;
  }

  .faq {
    margin-top: 44px;
  }
}

@media only screen and (max-width: 576px) {
  .footer {
    padding-top: 50px;

    &__main-title {
      font-size: 46px;
    }

    &__social-network-list-mobile {
      margin-left: 0;
      margin-right: 0;
    }
  }
}
</style>
