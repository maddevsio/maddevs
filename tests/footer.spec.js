import { shallowMount, createLocalVue } from '@vue/test-utils'
import Footer from '@/components/Footer'
import Vuelidate from 'vuelidate'

const localVue = createLocalVue()

localVue.use(Vuelidate)

describe('footer', () => {
  let wrapper

  beforeEach(() => {
    global.$nuxt = {
      $route: {
        name: '/',
      },
    }
    wrapper = shallowMount(Footer, {
      localVue,
    })
  })

  // ------ IMPORTANT ----- //
  it('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  // --------------------- //

  it('correctly sets the route when mounted', () => {
    expect(wrapper.vm.$data.currentPageName).toBe('/')
  })

  it('correctly sets the current pages name and class', () => {
    const $route = {
      path: '/gdpr',
    }

    wrapper = shallowMount(Footer, {
      mocks: {
        $route,
      },
    })

    wrapper.vm.$data.currentPageName = 'gdpr'
    expect(wrapper.vm.$route.path).toBe('/gdpr')
    expect(wrapper.vm.$data.currentPageName).toBe('gdpr')

    const wrapperClass = wrapper.findAll('.gdpr')
    expect(wrapperClass).toEqual({
      selector: '.gdpr',
    })
  })

  it('correctly sets the current pages name to /nda and class', () => {
    const $route = {
      path: '/nda',
    }

    wrapper = shallowMount(Footer, {
      mocks: {
        $route,
      },
    })

    wrapper.vm.$data.currentPageName = 'nda'
    expect(wrapper.vm.$route.path).toBe('/nda')
    expect(wrapper.vm.$data.currentPageName).toBe('nda')

    const wrapperClass = wrapper.findAll('.nda')
    expect(wrapperClass).toEqual({
      selector: '.nda',
    })
  })

  it('correctly sets the current pages name to /privacy and class', () => {
    const $route = {
      path: '/privacy',
    }

    wrapper = shallowMount(Footer, {
      mocks: {
        $route,
      },
    })

    wrapper.vm.$data.currentPageName = 'privacy'
    expect(wrapper.vm.$route.path).toBe('/privacy')
    expect(wrapper.vm.$data.currentPageName).toBe('privacy')

    const wrapperClass = wrapper.findAll('.privacy')
    expect(wrapperClass).toEqual({
      selector: '.privacy',
    })
  })
})
