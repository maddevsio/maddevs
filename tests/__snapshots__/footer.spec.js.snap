// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<footer
  class="footer /"
>
  <div
    class="container"
  >
    <div
      class="footer__content row"
    >
      <div
        class="footer__left-content col-xl-8 col-lg-7 col-md-6"
      >
        <h2
          class="footer__main-title"
        >
          Get In Touch
        </h2>
         
        <footercontacts-stub />
         
        <footernavbar-stub
          class="footer__navbar-desktop"
        />
      </div>
       
      <div
        class="footer__form-wrap col-xl-4 col-lg-5 col-md-5"
      >
        <footerform-stub />
      </div>
    </div>
     
    <div
      class="footer__mobile-content footer__social-network-list-mobile"
    >
      <footersocialnetworklist-stub />
    </div>
     
    <div
      class="footer__mobile-content"
    >
      <footernavbarmobile-stub />
    </div>
  </div>
</footer>
`;
